index.bs

```diff
@@ -3763,10 +3772,656 @@ nothrow>WritableStreamDefaultControllerError ( <var>controller</var>, <var>error
+  1. Set *this*.[[transformStreamController]] to _controller_.
+  1. Let _startPromise_ be <a>a new promise</a>.
+  1. Let _source_ be ! Construct(`<a idl>TransformStreamDefaultSource</a>`, « *this*, _startPromise_ »).
+  1. Let _readableStrategy_ be ! ObjectCreate(<a>%ObjectProperty%</a>).
+     <p class="note"><var>readableStrategy</var> is created from scratch so that the default <a>high water mark</a> can
+     be overridden. This does not apply to <var>writableStrategy</var>, so it is passed through unchanged from the
+     second constructor argument.</p>
+  1. Perform ! CreateDataProperty(_readableStrategy_, `"size"`, _size_).
+  1. Perform ! CreateDataProperty(_readableStrategy_, `"highWaterMark"`, _highWaterMark_).
+  1. Set *this*.[[readable]] to ? Construct(`<a idl>ReadableStream</a>`, « _source_, _readableStrategy_ »).
+  1. Let _sink_ be ! Construct(`<a idl>TransformStreamDefaultSink</a>`, « *this*, _startPromise_ »).
+  1. Set *this*.[[writable]] to ? Construct(`<a idl>WritableStream</a>`, « _sink_, _writableStrategy_ »).
+  1. Perform ! TransformStreamSetBackpressure(*this*, *true*).
+  1. Let _startResult_ be ? InvokeOrNoop(_transformer_, `"start"`, « _controller_ »).
+  1. Resolve _startPromise_ with _startResult_.
+</emu-alg>
+
+<h4 id="ts-prototype">Properties of the {{TransformStream}} Prototype</h4>
+
+<h5 id="ts-readable" attribute for="TransformStream" lt="readable">get readable</h5>
+
+<div class="note">
+  The <code>readable</code> getter gives access to the <a>readable side</a> of the transform stream.
+</div>
+
+<emu-alg>
+  1. If ! IsTransformStream(*this*) is *false*, throw a *TypeError* exception.
+  1. Return *this*.[[readable]].
+</emu-alg>
+
+<h5 id="ts-writable" attribute for="TransformStream" lt="writable">get writable</h5>
+
+<div class="note">
+  The <code>writable</code> getter gives access to the <a>writable side</a> of the transform stream.
+</div>
+
+<emu-alg>
+  1. If ! IsTransformStream(*this*) is *false*, throw a *TypeError* exception.
+  1. Return *this*.[[writable]].
+</emu-alg>
+
+<h3 id="ts-abstract-ops">General Transform Stream Abstract Operations</h3>
+
+<h4 id="is-transform-stream" aoid="IsTransformStream" nothrow>IsTransformStream ( <var>x</var> )</h4>
+
+<emu-alg>
+  1. If Type(_x_) is not Object, return *false*.
+  1. If _x_ does not have a [[transformStreamController]] internal slot, return *false*.
+  1. Return *true*.
+</emu-alg>
+
+<h4 id="transform-stream-error" aoid="TransformStreamError" nothrow>TransformStreamError ( <var>stream</var>, <var>e</var> )</h4>
+
+<emu-alg>
+  1. Perform ! WritableStreamDefaultControllerErrorIfNeeded(_stream_.[[writable]].[[writableStreamController]], _e_).
+  1. If _stream_.[[readable]].[[state]] is `"readable"`, perform !
+     ReadableStreamDefaultControllerError(_stream_.[[readable]].[[readableStreamController]], _e_).
+  1. If _stream_.[[backpressure]] is *true*, perform ! TransformStreamSetBackpressure(_stream_, *false*).
```

Comment:
This has a comment in the source that seems like it might be useful to include in the standard as a note. However it references things I'm not sure about, like write()/pull()/enqueue(). Are those on the transform stream source/sink? Maybe it can be rephrased to capture the important points without the details; not sure.